frontend/src/components/widgets/behavior/FloatingMenuToggleGroup.vue

```diff
@@ -0,0 +1,38 @@
+<template>
+	<slot></slot>
+</template>
+
+<script lang="ts">
+import { defineComponent, toRef, computed } from "vue";
+import FloatingMenu from "@/components/widgets/floating-menus/FloatingMenu.vue";
+
+/**
+ * Transparent component that groups together multiple togglable floating menus such that the user
+ * can easily move between different menus by hovering their mouse over other toggles.
+ *
+ * Behavior-only component. No props are necessary. No HTML is added to the DOM.
+ */
+export default defineComponent({
+	data() {
+		return {
+			openMenu: undefined as typeof FloatingMenu | undefined,
+		};
+	},
+	provide() {
+		// Injected by FloatingMenuToggleButton and FloatingMenu, where grouping behavior is implemented.
+		return {
+			openedMenuInGroup: toRef(this, "openMenu"),
+			notifyGroupedFloatingMenuOpened: (newOpenMenu: typeof FloatingMenu) => {
+				this.openMenu = newOpenMenu;
+			},
+			notifyGroupedFloatingMenuClosed: (closedMenu: typeof FloatingMenu) => {
+				if (this.openMenu === closedMenu) {
+					this.openMenu = undefined;
+				}
+			},
+			// Creates a new computed variable, similar to one created in the computed section of a component.
+			openMenuOnHover: computed(() => this.openMenu !== undefined),
+		};
+	},
+});
```

Comment:
Can you move all this logic into a new file in `frontend/src/utilities` so we are consistent in the way we use `provide/inject`? See `dialog.ts`, `documents.ts`, and `fullscreen.ts` for how that's done. Then we can import those functions into this Vue component. Being consistent about the way we use that logic would be helpful for understanding (I was confused for several minutes trying to understand this and how it related to the previously established pattern of `provide/inject` usage).